buildchain/config.py

```diff
@@ -0,0 +1,40 @@
+# coding: utf-8
+
+
+"""Build configuration parameters.
+
+Those can be edited by the user to customize/configure the build system.
+"""
+
+
+from typing import Tuple
+from pathlib import Path
+
+from buildchain import ROOT
+
+
+# Project name.
+PROJECT_NAME : str = 'MetalK8s'
+
+# Path to the root of the build directory.
+BUILD_ROOT : Path = ROOT/'_build'
+
+# Vagrant configuration.
+VAGRANT_PROVIDER : str = 'virtualbox'
```

Comment:
Is there a way to override this value without changing any Git-tracked files?

Reply:
Not yet (same as the Makefile approach).

That's why I suggested/asked about supporting a `build.ini` configuration file that could be used for that purpose.
Seeing this comment I'll consider that we want to go that way and create an issue to track this task (or maybe do it in this PR if it's small enough).

Reply:
```make
VAGRANT_DEFAULT_PROVIDER = virtualbox
VAGRANT_UP_OPTS ?= --provision --no-destroy-on-error --parallel --proider=$(VAGRANT_DEFAULT_PROVIDER)
```

so either `make vagrantup VAGRANT_DEFAULT_PROVIDER=libvirt` or an environment would work ;-) Could argue the variable name could be improved upon.

Reply:
Not overly fond of some `config.ini` file. That's hard to keep in sync, or to set up in CI.

Reply:
Can you detail a little bit? What kind of sync issues?

Other than a config file, what could we use? Environment variables (maybe easier to use in CI, less so for a human user though)? Something even better :-)?

Reply:
> Can you detail a little bit? What kind of sync issues?

Do you check-in a config `config.ini`, which would then confict when people make local modifications? A `config.ini.example`? ...

For humans, one can use `dotenv` or similar :)

Reply:
> Do you check-in a config `config.ini`, which would then confict when people make local modifications? A `config.ini.example`? ...

That was the idea, yeah (a `config.ini.example`).

Is something wrong with that?

Didn't know about `dotenv`, looks nice but introduces a new dependency it seems. What's the avantage over a `.ini`?

Reply:
Done, with a `.env` at the root of the repo.